src/coreclr/jit/lsraarm64.cpp

```diff
@@ -1634,15 +1635,32 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree, int* pDstCou
                         {
                             predMask = RBM_LOWMASK.GetPredicateRegSet();
                         }
+
+                        // Special-case, CreateBreakPropagateMask's op2 is the RMW node.
+                        if (intrinEmb.id == NI_Sve_CreateBreakPropagateMask)
+                        {
+                            assert(embOp2Node->isRMWHWIntrinsic(compiler));
+                            assert(!tgtPrefOp1);
+                            assert(!tgtPrefOp2);
+                            assert(!tgtPrefEmbOp2OfOp2);
+                            tgtPrefEmbOp2OfOp2 = true;
```

Comment:
nit: Long term (I don't think it needs to be done in this PR) it'd probably be beneficial to have some `GetRmwOpNum` helper so we're not assuming its always `op1`.

We have something similar in the form of `GetResultOpNumForRmwIntrinsic` which is used for `RMW` intrinsics like `FusedMultiplyAdd` where the operands can be swapped to any position as multiple instruction encodings exist and so its basically determining which operand is the "optimal" one to pick as the RMW operand.

While this `GetRmwOpNum` concept would be saying, what is statically the RMW operand assuming that no further transforms will happen and so would statically say `2` here for a case like `NI_Sve_CreateBreakPropagateMask` and may statically say `1` for other scenarios.

Reply:
I agree with this. The current logic is getting a little too complicated and I can imagine it growing in the future. Instead of it making assumptions, the API needs to tell us which node is the RMW node and LSRA handles it accordingly.